scripts/run_tests.sh

```diff
@@ -6,18 +6,27 @@ set -e
 (cd /app/relation_engine_server/test/spec_release && \
   tar czvf spec.tar.gz sample_spec_release)
 
-black .
-flake8 --max-complexity 20 /app
-mypy --ignore-missing-imports /app
-bandit -r /app
+# Run code quality checks before tests; a failure here
+# will prevent the tests from running.
+source_dirs="/app/relation_engine_server /app/client_src /app/spec /app/common"
+flake8 --max-complexity 20 $source_dirs
+mypy --ignore-missing-imports $source_dirs
+bandit -r $source_dirs
 
 # start server, using the specs in /spec/repo
 sh /app/scripts/start_server.sh &
 coverage erase
+
 # spec validation
 python -m spec.validate
+
 # run importer/, relation_engine_server/, and spec/ tests
-coverage run --branch -m unittest discover -v
+coverage run --branch --parallel-mode -m unittest discover -v importers
+coverage run --branch --parallel-mode -m unittest discover -v relation_engine_server
+coverage run --branch --parallel-mode -m unittest discover -v spec
```

Comment:
why did this change?

Reply:
I don't remember if this was to specifically to avoid tests picking up a venv directory or the test directory, but generally to constrain the tests to the specified directories without any surprises; they are separate invocations because discover allows a single root directory (I'm pretty sure I would have tried that first!)

Reply:
Cool, thanks for the explanation!